examples/sampling/node_classification.py

```diff
@@ -0,0 +1,284 @@
+        # Enable pin_memory for faster CPU to GPU data transfer if the
+        # model is running on a GPU.
+        pin_memory = buffer_device != device
+
+        for layer_idx, layer in enumerate(self.layers):
+            is_last_layer = layer_idx == len(self.layers) - 1
+            y = torch.empty(
+                g.num_nodes(),
+                self.out_size if is_last_layer else self.hidden_size,
+                device=buffer_device,
+                pin_memory=pin_memory,
+            )
+            feat = feat.to(device)
+            for input_nodes, output_nodes, blocks in tqdm.tqdm(dataloader):
+                x = feat[input_nodes]
+                hidden_x = layer(blocks[0], x)  # len(blocks) = 1
+                if layer_idx != len(self.layers) - 1:
+                    hidden_x = F.relu(hidden_x)
+                    hidden_x = self.dropout(hidden_x)
+                # By design, our output nodes are contiguous.
+                y[output_nodes[0] : output_nodes[-1] + 1] = hidden_x.to(
+                    buffer_device
+                )
+            feat = y
+        return y
+
+
+@torch.no_grad()
+def evaluate(model, graph, dataloader, num_classes):
+    model.eval()
+    ys = []
+    y_hats = []
+    for it, (input_nodes, output_nodes, blocks) in enumerate(dataloader):
+        x = blocks[0].srcdata["feat"]
+        ys.append(blocks[-1].dstdata["label"])
+        y_hats.append(model(blocks, x))
+    return MF.accuracy(
+        torch.cat(y_hats),
+        torch.cat(ys),
+        task="multiclass",
+        num_classes=num_classes,
+    )
+
+
+@torch.no_grad()
+def layerwise_infer(device, graph, nid, model, num_classes, batch_size):
+    model.eval()
+    pred = model.inference(graph, device, batch_size)  # pred in buffer_device.
+    pred = pred[nid]
+    label = graph.ndata["label"][nid].to(pred.device)
+    return MF.accuracy(pred, label, task="multiclass", num_classes=num_classes)
+
+
+def train(args, device, g, dataset, model, num_classes, use_uva):
+    # Create sampler & dataloader.
+    train_idx = dataset.train_idx.to(device)
+    val_idx = dataset.val_idx.to(device)
+    #####################################################################
+    # (HIGHLIGHT) Instantiate a NeighborSampler object for efficient
+    # training of Graph Neural Networks (GNNs) on large-scale graphs.
+    #
+    # The argument [10, 10, 10] sets the number of neighbors (fanout)
+    # to be sampled at each layer. Here, we have three layers, and
+    # 10 neighbors will be randomly selected for each node at each
+    # layer.
+    #
+    # The 'prefetch_node_feats' and 'prefetch_labels' parameters
+    # specify the node features and labels that need to be pre-fetched
+    # during sampling. Pre-fetching data in this way is advantageous
+    # as it can reduce the I/O overhead during the training process,
+    # making the whole computation more efficient. In this case,
+    # the feature 'feat' and the label 'label' will be pre-fetched.
```

Comment:
```
Pre-fetching data in this way is advantageous
    # as it can reduce the I/O overhead during the training process,
    # making the whole computation more efficient. In this case,
    # the feature 'feat' and the label 'label' will be pre-fetched.
```

This part does not add enough information as I can easily get the point that the prefetch will reduce I/O overhead in someway, but it is more helpful to elaborate how prefetching works.